xformers/components/attention/csrc/cuda/mem_eff_attention/kernel_forward.h

```diff
@@ -88,6 +102,7 @@ struct AttentionKernel {
     scalar_t* query_ptr; // [num_queries, num_heads, head_dim]
     scalar_t* key_ptr; // [num_keys, num_heads, head_dim]
     scalar_t* value_ptr; // [num_keys, num_heads, head_dim_value]
+    scalar_t* attn_bias_ptr = nullptr;  // [num_heads, num_queries, num_keys]
```

Comment:
(You might not know the answer, but) I'm wondering how much we could gain by having:
(1) The bias already in the right format (eg if you know you want to use this bias only with MHA, you could store it in a format easy to load from gmem directly, without having to go through shared-memory - this format would be different depending on the kernel running tho)
(2) A boolean mask or some datatype with even lower precision

Reply:
applied after 1st GEMM (Q @ K.T) and its 2d matrix

Reply:
if it is a 2D matrix, using shared memory may not be bad.  if you load directly, you can only load 2 elements a time, it is not the most efficient way for the memory BW.  If you transform through the shared memory, you can load 128bit data a time to fully use the memory BW